app/models/links.rb

```diff
@@ -0,0 +1,69 @@
+class Links
+  # Links available for indexed gems
+  LINKS = {
+    'home'     => 'homepage_uri',
+    'code'     => 'source_code_uri',
+    'docs'     => 'documentation_uri',
+    'wiki'     => 'wiki_uri',
+    'mail'     => 'mailing_list_uri',
+    'bugs'     => 'bug_tracker_uri',
+    'download' => 'download_uri'
+  }.freeze
+
+  # Links available for non-indexed gems
+  NON_INDEXED_LINKS = {
+    'docs'     => 'documentation_uri'
+  }.freeze
+
+  attr_accessor :rubygem, :version, :linkset
+
+  def initialize(rubygem = nil, version = nil)
+    self.rubygem = rubygem
+    self.version = version
+    self.linkset = rubygem.linkset
+  end
+
+  def links
+    version.indexed ? LINKS : NON_INDEXED_LINKS
+  end
+
+  delegate :keys, to: :links
+
+  def each
+    return enum_for(:each) unless block_given?
+    links.each do |short, long|
+      value = send(long)
+      yield short, value if value
+    end
+  end
+
+  # documentation url:
+  # if metadata has it defined, use that
+  # or if linksets has it defined, use that
+  # else, generate one from gem name and version number
+  def documentation_uri
+    (version.metadata && version.metadata["documentation_uri"].presence) ||
```

Comment:
@sonalkr132 Looks like `Version#metadata` will have a hash. Made that change.

I could have sworn I tried this a number of times with the previous PR, but the tests would fail. Looks like it passes now.